compiler/rustc_span/src/def_id.rs

```diff
@@ -369,7 +369,7 @@ rustc_data_structures::define_id_collections!(DefIdMap, DefIdSet, DefIdMapEntry,
 /// few cases where we know that only `DefId`s from the local crate are expected;
 /// a `DefId` from a different crate would signify a bug somewhere. This
 /// is when `LocalDefId` comes in handy.
-#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+#[derive(Clone, Copy, PartialEq, Eq, Hash, TypeFoldable, TypeVisitable)]
```

Comment:
why is that needed?

Reply:
As with `DefId`, it appears in anonymous tuple arguments to `ParamEnv::and`.

Reply:
Replaced with named structs in 1eda0645a07d1adc94eb9c25b9bf5d110ae31639